### PR TITLE
Implement API Server

### PR DESCRIPTION
**Pull request by hochiw**
- /healthz
- /register
- /deregister
- /collections
- /collections/:blockchain/:network/owners/:contract_address
- /nfts/:owner_address

Postman collection
[AuthGear NFT Indexer API Server.postman_collection.json.zip](https://github.com/authgear/authgear-nft-indexer/files/9348644/AuthGear.NFT.Indexer.API.Server.postman_collection.json.zip)

refs #2

### CODE REVIEW

**Review by louischan-oursky — COMMENTED** [review]
```diff
diff --git a/pkg/mutator/nft_collection.go b/pkg/mutator/nft_collection.go
index 774a384..695f62e 100644
--- a/pkg/mutator/nft_collection.go
+++ b/pkg/mutator/nft_collection.go
@@ -17,11 +17,11 @@ type NFTCollectionMutator struct {
 
 func (q *NFTCollectionMutator) InsertNFTCollection(blockchainNetwork model.BlockchainNetwork, contractName string, contractAddress string) (*eth.NFTCollection, error) {
 	collection := &eth.NFTCollection{
-		Blockchain:        blockchainNetwork.Blockchain,
-		Network:           blockchainNetwork.Network,
-		ContractAddress:   contractAddress,
-		Name:              contractName,
-		SyncedBlockHeight: *bunbig.FromInt64(0),
+		Blockchain:      blockchainNetwork.Blockchain,
+		Network:         blockchainNetwork.Network,
+		ContractAddress: contractAddress,
+		Name:            contractName,
+		FromBlockHeight: *bunbig.FromInt64(0),
 	}
 
 	err := q.Session.RunInTx(q.Ctx, nil, func(ctx context.Context, tx bun.Tx) error {

```

File: pkg/query/nft_owner.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+type NFTOwnerQuery struct {
+	Ctx     context.Context
+	Session *bun.DB
+}
+
+type NFTOwnerQueryBuilder struct {
+	*bun.SelectQuery
+}
+
+func (b NFTOwnerQueryBuilder) WithBlockchainNetwork(blockchainNetwork model.BlockchainNetwork) NFTOwnerQueryBuilder {
```
[the remark this answers is not in the document]
Comment (hochiw) [reply pkg/query/nft_owner.go:21]: Seems it cannot inherit the underlying function declarations

```
b.Where undefined (type *NFTOwnerQueryBuilder has no field or method Where)
```

**Comment by hochiw** [comment]
@louischan-oursky Updated :adore2x:

File: pkg/web3/alchemy.go [hunk 2]
```diff
@@ -63,26 +70,47 @@ func (a *AlchemyAPI) GetNFTTransfers(blockchainNetwork model.BlockchainNetwork,
 		return nil, fmt.Errorf("failed to marshal json: %s", err)
 	}
 
-	log.Printf("Requesting NFT Transfers for contractAddresses: %s from network %s, fromBlock %s, toBlock %s from endpoint %s", strings.Join(contractAddresses, ", "), alchemyNetwork.Network, fromBlock, toBlock, alchemyNetwork.Endpoint)
-	res, err := http.Post(alchemyNetwork.Endpoint+alchemyNetwork.APIKey, "application/json", bytes.NewBuffer(jsonBody))
+	log.Printf("Requesting NFT Transfers for contractAddresses: %s from network %s %s, fromBlock %s, toBlock %s", strings.Join(contractAddresses, ", "), blockchainNetwork.Blockchain, blockchainNetwork.Network, fromBlock, toBlock)
+	res, err := http.Post(alchemyEndpoints.TransferEndpoint, "application/json", bytes.NewBuffer(jsonBody))
 	if err != nil {
 		return nil, fmt.Errorf("failed to send request: %s", err)
```
[the remark this answers is not in the document]
Comment (louischan-oursky) [reply pkg/web3/alchemy.go]: Turn on `errorlint` in golangci-lint. Use `%+v` to log error, use `%w` to wrap error.

File: pkg/web3/alchemy.go [hunk 3]
```diff
@@ -19,20 +18,28 @@ type AlchemyAPI struct {
...
+	NFTEndpoint      string
+}
+
+func (a *AlchemyAPI) getRequestEndpoints(blockchainNetwork model.BlockchainNetwork) *AlchemyEndpoint {
 	alchemyNetworks := a.Config.Alchemy
 	for _, alchemyNetwork := range alchemyNetworks {
 		if alchemyNetwork.Blockchain == blockchainNetwork.Blockchain && alchemyNetwork.Network == blockchainNetwork.Network {
-			return &alchemyNetwork
+			return &AlchemyEndpoint{
+				TransferEndpoint: fmt.Sprintf("%s/v2/%s", alchemyNetwork.Endpoint, alchemyNetwork.APIKey),
```
Comment (louischan-oursky) [inline pkg/web3/alchemy.go:31]: Use `path` package to join path. Use `net/url` to add path to endpoint.

```golang
endpoint, _ := url.Parse(alchemyNetwork.Endpoint)
endpoint.Path = path.Join("/", "v2", alchemyNetwork.APIKey)
```

**Comment by hochiw** [comment]
@louischan-oursky updated and tested :adore2x:

**Pull request merged**